cmd/snap/cmd_validate.go

```diff
@@ -0,0 +1,193 @@
+
+func splitValidationSetArg(arg string) (account, name string, seq int, err error) {
+	parts := strings.Split(arg, "=")
+	if len(parts) > 2 {
+		return "", "", 0, fmt.Errorf("cannot parse validation set, expected account/name=seq")
+	}
+	if len(parts) == 2 {
+		seq, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return "", "", 0, err
+		}
+	}
+
+	parts = strings.Split(parts[0], "/")
+	if len(parts) != 2 {
+		return "", "", 0, fmt.Errorf("expected a single account/name")
+	}
+
+	account = parts[0]
+	name = parts[1]
+	if !validName.MatchString(account) {
+		return "", "", 0, fmt.Errorf("invalid account name %q", account)
+	}
+	if !validName.MatchString(name) {
+		return "", "", 0, fmt.Errorf("invalid name %q", name)
+	}
+
+	return account, name, seq, nil
+}
+
+func fmtValid(res *client.ValidationSetResult) string {
+	if res.Valid {
+		return fmt.Sprintf("valid")
+	}
+	return fmt.Sprintf("invalid")
+}
+
+func (cmd *cmdValidate) Execute(args []string) error {
+	// check that only one action is used at a time
+	var action string
+	for _, a := range []struct {
+		name string
+		set  bool
+	}{
+		{"monitor", cmd.Monitor},
+		{"enforce", cmd.Enforce},
+		{"forget", cmd.Forget},
+	} {
+		if a.set {
+			if action != "" {
+				return fmt.Errorf("cannot use --%s and --%s together", action, a.name)
+			}
+			action = a.name
+		}
+	}
+
+	if cmd.Positional.ValidationSet == "" && action != "" {
+		return fmt.Errorf("missing validation set argument")
+	}
+
+	var account, name string
+	var seq int
+	var err error
+	if cmd.Positional.ValidationSet != "" {
+		account, name, seq, err = splitValidationSetArg(cmd.Positional.ValidationSet)
+		if err != nil {
+			return fmt.Errorf("cannot parse validation set %q: %v", cmd.Positional.ValidationSet, err)
+		}
+	}
+
+	if action != "" {
+		// forget
+		if cmd.Forget {
+			return cmd.client.ForgetValidationSet(account, name, seq)
+		}
+		// apply
+		opts := &client.ValidateApplyOptions{
+			Mode:     action,
+			Sequence: seq,
+		}
+		return cmd.client.ApplyValidationSet(account, name, opts)
+	}
+
+	// no validation set argument, print list with extended info
+	if cmd.Positional.ValidationSet == "" {
+		vsets, err := cmd.client.ListValidationsSets()
+		if err != nil {
+			return err
+		}
+		if len(vsets) == 0 {
+			return nil
+		}
+
+		esc := cmd.getEscapes()
+		w := tabWriter()
+
+		// TRANSLATORS: the %s is to insert a filler escape sequence (please keep it flush to the column header, with no extra spaces)
+		fmt.Fprintf(w, i18n.G("Validation\tMode\tSeq\tCurrent\t%s\tNotes\n"), fillerPublisher(esc))
+		for _, res := range vsets {
+			// TODO: fill notes when've clarity about them
+			var notes string
+			// doing it this way because otherwise it's a sea of %s\t%s\t%s
+			line := []string{
+				res.ValidationSet,
+				res.Mode,
+				fmt.Sprintf("%d", res.Sequence),
+				fmtValid(res),
+				notes,
+			}
+			fmt.Fprintln(w, strings.Join(line, "\t"))
+		}
+		w.Flush()
+		fmt.Fprintln(Stdout)
```

Comment:
We use fmt.Fprintln() above, why an extra fmt.Fprintln() here? Is that needed?

Reply:
Hmm, maybe not, we are probably not super consistent. We do that also in 'snap changes' for an extra blank line after all the output. But this seems to be the only cases (apart from some snap debug commands), so I'm going to remove it. Thanks!